src/legacy/ui/public/chrome/api/sub_url_hooks_utils.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function areDecodedHashesEqual(urlA: string, urlB: string): boolean {
+  const getHash = (url: string) => url.split('#')[1] ?? '';
+  const hashA = getHash(urlA);
+  const decodedHashA = decodeURIComponent(hashA);
+
+  const hashB = getHash(urlB);
+  const decodedHashB = decodeURIComponent(hashB);
+
+  return hashA !== hashB && decodedHashA === decodedHashB;
```

Comment:
NIT: name of the function a little misleading, but I guess `areHashesDistinctsButDecodedHashesEquals` isn't better...